Sitefinity CLI/Commands/UpgradeCommand.cs

```diff
@@ -51,447 +45,97 @@ internal class UpgradeCommand
-                            if (lastProgressUpdate != progressInfo)
-                            {
-                                lastProgressUpdate = progressInfo;
-                                this.logger.LogInformation(progressInfo);
-                            }
-                        }
-                        catch { }
-                    }
-
-                    iterations++;
-
-                    // Last operation is still not executed and the file is not created.
-                    System.Threading.Thread.Sleep(waitStep);
-                    continue;
-                }
-
-                System.Threading.Thread.Sleep(waitStep);
-                var result = this.ReadAllTextFromFile(resultFile);
-                if (result != "success")
-                {
-                    this.logger.LogError(string.Format("Error occured while upgrading nuget packages. {0}", result));
-                    throw new UpgradeException("Upgrade failed");
-                }
-
-                break;
-            }
-
-            this.logger.LogInformation("Operation completed successfully!");
-        }
-
-        private string ReadAllTextFromFile(string path)
-        {
-            using (var fileStream = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
-            using (var textReader = new StreamReader(fileStream))
-            {
-                return textReader.ReadToEnd();
-            }
-        }
-
-        private void SyncProjectReferencesWithPackages(IEnumerable<string> projectFilePaths, string solutionFolder)
-        {
-            foreach (string projectFilePath in projectFilePaths)
-            {
-                this.sitefinityPackageManager.SyncReferencesWithPackages(projectFilePath, solutionFolder);
-            }
-        }
-
-        private IList<string> GetProjectsPathsFromSolution(string solutionPath, bool onlySitefinityProjects = false)
-        {
-            if (!solutionPath.EndsWith(Constants.SlnFileExtension, StringComparison.InvariantCultureIgnoreCase))
-            {
-                throw new UpgradeException(string.Format(Constants.FileIsNotSolutionMessage, solutionPath));
-            }
-
-            IEnumerable<string> projectFilesAbsolutePaths = SolutionFileEditor.GetProjects(solutionPath)
-                .Select(sp => sp.AbsolutePath)
-                .Where(ap => (ap.EndsWith(Constants.CsprojFileExtension, StringComparison.InvariantCultureIgnoreCase) || ap.EndsWith(Constants.VBProjFileExtension, StringComparison.InvariantCultureIgnoreCase)));
-
-            if (onlySitefinityProjects)
-            {
-                projectFilesAbsolutePaths = projectFilesAbsolutePaths
-                    .Where(ap => this.HasSitefinityReferences(ap) && this.HasValidSitefinityVersion(ap));
-            }
-
-            return projectFilesAbsolutePaths.ToList();
-        }
-
-        private bool HasValidSitefinityVersion(string projectFilePath)
+        private async Task Validate()
```

Comment:
There are several propblems in the Validate method. on Line 117 where we  if (!this.SkipPrompts && !this.promptService.PromptYesNo(Constants.UpgradeWarning)) we cancel the execution of the method with the return statement. However this will not cancel the exection of the command like it did before.
Also in the validate method we set the version property if it is not set. This is something odd. An method should do one thing and should not have such side effects